### ISSUE TITLE
`reportMissingSuperCall` shouldn't report an error on classes that extend `AbstractContextManager`

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#6626]
**Is your feature request related to a problem? Please describe.**
when extending `AbstractContextManager`, it's safe to not call the `super().__enter__` and `super().__exit__` methods because they're abstract and don't do anything

```py
from contextlib import AbstractContextManager
from typing import override, final

@final
class SuppressError(AbstractContextManager[None]):
    @override
    def __enter__(self) -> None: # error: Method "__enter__" does not call the method of the same name in parent class
        ...
```

**Describe the solution you’d like**
- don't report the errors on the `__enter__` and `__exit__` methods if the super method is abstract
- if this is the same issue as https://github.com/microsoft/pyright/issues/4893#issuecomment-1494876640, then it should not report an error when the class is decorated with `@final`

**Comment by erictraut** [comment]
The `reportMissingSuperCall` is working as designed. I don't plan to add a bunch of special cases for this check. If you want to use this check, you can suppress it where you see fit.
